src/nvapi_d3d12.cpp

```diff
@@ -285,8 +285,50 @@ extern "C" {
         // Note: One might think that SM here is D3D12 Shader Model, in fact it is the "Streaming Multiprocessor" architecture version
         // Values are valid for Turing and newer only, due to VK_NV_cuda_kernel_launch not being supported by earlier generations
         auto computeCapability = adapter->GetComputeCapability();
-        pGraphicsCaps->majorSMVersion = computeCapability.first;
-        pGraphicsCaps->minorSMVersion = computeCapability.second;
+        if (computeCapability.has_value()) {
+            pGraphicsCaps->majorSMVersion = computeCapability.value().first;
+            pGraphicsCaps->minorSMVersion = computeCapability.value().second;
+        } else if (adapter->HasNvProprietaryDriver()) {
```

Comment:
`|| HasNvk()` perhaps? IIRC we didn't have any driver checks previously here but I'm also not sure what behavior would be better for NVK.

Reply:
Good question. I tend to report `0` for NVK like we currently do (because no `VK_NV_cuda_kernel_launch` for NVK). I guess it also doesn't matter that much, at least the original PR for DLLS/D3D12 didn't set those values at all (https://github.com/jp7677/dxvk-nvapi/pull/32) when introducing this entry point.